elastic/kibana
# [Stack Monitoring] Update security messaging around alerting

## Summary 

Fixes #115236

Alerting no longer requires TLS. However, it does still require API keys to be enabled, and an encryption key to be set. This PR aligns the messaging with the changes in https://github.com/elastic/kibana/pull/115234.

I've also changed the way that Stack Monitoring decides "is this secure?". 

Previously the alerting logic was here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/routes/health.ts#L51 and here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/routes/legacy/health.ts#L38 

And then Stack Monitoring was doing it's own check here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/server/lib/elasticsearch/verify_alerting_security.ts#L28 (with slightly different logic).

I've added a `getSecurityHealth` method on the Alerting setup contract so that Alerting can be the source of truth for this, and Stack Monitoring can stop running it's own independent check. I've also updated the two alerting routes to use that helper function.

Alerting Team: please let me know if you think this is a bad idea.

## Testing

We can alter the testing steps from https://github.com/elastic/kibana/pull/115234#issue-1027639899.

Explicitly disabling API keys
1. `yarn es snapshot -E xpack.security.authc.api_key.enabled=false` and `yarn start`
2. Head to Stack Monitoring and click Alerts and Rules > Create default rules
3. Observe the toast wording

Ephemeral encryption key + disabling API keys
1. Change https://github.com/elastic/kibana/blob/master/x-pack/plugins/encrypted_saved_objects/server/plugin.ts#L61 to `const canEncrypt = false;`
2. Head to Stack Monitoring and click Alerts and Rules > Create default rules
3. Observe the toast wording

 ## 🎨  New wording:

<img width="402" alt="Screenshot 2021-11-04 at 14 13 14" src="https://user-images.githubusercontent.com/471693/140329419-ae9ea2ed-c830-4758-92fb-8f71655d0fc7.png">
